source/custom/nidcpower_service.custom.cpp

```diff
@@ -55,4 +55,18 @@ ::grpc::Status NiDCPowerService::MeasureMultiple(::grpc::ServerContext* context,
     }
   }
 
+::grpc::Status NiDCPowerService::ConvertApiErrorStatusForViSession(google::protobuf::int32 status, ViSession vi)
+{
+    const ViInt32 buffer_size = 4096;
+    ViStatus error_code {};
+    std::string description(buffer_size, '\0');
+    // Try first to get the most recent error with a dynamic message.
+    library_->GetError(vi, &error_code, buffer_size, description.data());
+    if (error_code != status) {
+        // Since another thread has changed the status, fall back to the static message lookup.
+        library_->ErrorMessage(vi, status, description.data());
```

Comment:
This function (and many of the similar MI error_message functions) assumes the buffer size is >= 256. That assumption is currently true, but consider documenting it.

Reply:
Added `static_assert` wherever the function signature indicated that the buffer length was assumed, such as `char message[256]`.